Dz2

devAlexanderKonopelko opened the pull request:
In this task I worked with Layouts to build Flags. Also the cat animation was created)
To see the task done click the See Second Task button.

devAlexanderKonopelko committed:
Home task #2: Flags and Animation

Worked with Layouts, its properties. Made an animation.

devAlexanderKonopelko committed:
Create AnimationActivity.kt

devAlexanderKonopelko committed:
Update AndroidManifest.xml

akelian reviewed (approved):
Approved

ciszkin commented:
Cool stuff with TextViews as stripes of flags!

ciszkin reviewed (approved):
Good job!

TMSlebedev commented on FirstApp/app/src/main/java/com/example/firstapp/AnimationActivity.kt:
```diff
@@ -0,0 +1,34 @@
+            val animation = AnimationDrawable()
+            animation.isOneShot = false
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+                animation.addFrame(getDrawable(R.drawable.image0)!!, 250)
```
Кто-то снова любит получать NPE?

TMSlebedev commented on FirstApp/app/src/main/java/com/example/firstapp/AnimationActivity.kt:
```diff
@@ -0,0 +1,34 @@
+
+            val animation = AnimationDrawable()
+            animation.isOneShot = false
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
```
А если версия ниже LOLLIPOP, то не судьба увидеть анимацию?

TMSlebedev commented on FirstApp/app/src/main/res/drawable/animation_cat.xml:
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<animation-list
```
Зачем тогда в коде добавлять кадры, если есть эта xml-ка? Почему бы не использовать ее?

TMSlebedev reviewed (commented):
Исправить замечания

devAlexanderKonopelko committed:
Animation changes

Changed  the way animation works, fixed some version problems which might appear)
